docs/src/ref_impl/creating_appvm.md

```diff
@@ -0,0 +1,90 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Creating an Application VM
+
+## What is AppVM?
+
+AppVM is a virtual machine, which isolates an application from a host OS. An application seems like it is running inside an ordinary OS, yet it runs in a completely different system. An application inside renders, shows and controls from a host OS via GUI VM, network access is provided by NetVM.
+
+In the perfect scenario, there is a separate AppVM for every application. But it is possible to run multiple applications inside one AppVM.
```

Comment:
We could add here also the rationale for the AppVM - something like 

```
## Why AppVM?

AppVMs are used to improve trust in system components by isolating the apps from both host OS and other applications. This way user can use applications of different trust levels within the same system and without compromising system security. This is because virtualization with hardware backed mechanisms provides better resource protection than traditional OS. While the VMs have overhead, it's acceptable via improved security and usability that makes the application seem like it is running inside an ordinary OS.

As a result - both highly trusted applications and untrusted applications can be hosted in the same secure system when the concerns are separated in their own AppVMs.
```

Reply:
Added